test/test_models.py

```diff
@@ -40,22 +52,35 @@ def get_available_video_models():
 # they are not yet supported in JIT.
 script_test_models = [
     "deeplabv3_resnet101",
-    "mobilenet_v2",
-    "resnext50_32x4d",
     "fcn_resnet101",
-    "googlenet",
-    "densenet121",
-    "resnet18",
-    "alexnet",
-    "shufflenet_v2_x1_0",
-    "squeezenet1_0",
-    "vgg11",
-    "inception_v3",
     'r3d_18',
 ]
 
 
 class ModelTester(TestCase):
+    # set random seed for whatever callable follows (if any)
+    # can be called with no args to just set to standard random seed
+    def _rand_sync(self, callable=None, **kwargs):
```

Comment:
I would find it more natural (and pythonic) to have either a context manager that sets the seed (and upon exit restore its initial value) or just call `set_rng_seed(STANDARD_SEED)` in the call sites of `_rand_sync` instead of having this function.

I think what annoys me is that the function seems to isolate the rng, but it doesn't actually restore it to its initial value, so I think the function is misleading in its behavior and has side-effects that might not be expected by the user.

Reply:
I took this approach because the only way to ensure determinism with arbitrary test logic (i.e., with the developer having the freedom to add or remove operations that hit the RNG) is to make all RNG-related operations atomic, and perform them from a known state - i.e., seeding explicitly every time. Anything else means that sometimes you will have to regenerate values for the correctness tests. I've tried to hide the mechanics of that away from top-level tests, instead bundling the RNG seeding with each (atomic) operation.

Reply:
As the code currently stands, we only use `_rand_sync` in two places: `_get_test_input` and `_get_test_model`, and we could add the `set_rng_seed` in both locations. Do you plan to use it in other places as well?

Reply:
`_rand_sync()` actually pre-dated `set_rng_seed()` - I changed the former to call the latter after it was added, to limit duplication. I can switch everything to call `set_rng_seed()` if that resolves the conversation.